python/cudf/cudf/core/column/column.py

```diff
@@ -179,9 +179,11 @@ def any(self):
         return bool(libcudf.reduce.reduce("any", self, dtype=np.bool_))
 
     def __sizeof__(self):
-        n = self.data.size
+        n = self.data.size if self.data is not None else 0
         if self.nullable:
             n += self.mask.size
```

Comment:
If `self.offset` or `self.size` is set due to slicing the Column these numbers wont be representative of the actual memory usage of the Column. My thoughts are we should use `self.base_data.size` and `self.base_mask.size` instead.

Reply:
dask-cudf has a unittest that assumes that cudf.DataFrame.memory_usage returns the size of the slice rather than of the underlying dataframe - and this change broke that test =(. 

I fixed the test in the latest commit (https://github.com/rapidsai/cudf/pull/6549/commits/f48903e5a72a8d09144cdd072d22f99392bf6349 ) by forcing a copy of each partition, but I'm worried that this might lead to some unexpected behaviour (like if the dask dataframe is constructed from a cudf dataframe by calling ```dask.dataframe.from_pandas(gdf, n_partitions=10)``` calling memory_usage on this dask dataframe will return 10x the actual memory usage).